array/builder.gen.go

```diff
@@ -79,6 +79,18 @@ func (b *IntBuilder) CopyValidValues(values *Int) {
 		}
 	}
 }
+func (b *IntBuilder) CopyValuesWithMask(values *Int, mask Array) {
+	b.Reserve(mask.Len() - mask.NullN())
+	for i := 0; i < values.Len(); i++ {
+		if mask.IsValid(i) {
```

Comment:
Seems we had the same issue though I tweaked the existing `CopyValidValues` function in https://github.com/influxdata/flux/pull/5149 . I also ensured that `mask.IsValid` does not go through in indirect call since this code is pretty hot.